File: 8kyu/(8 kyu) Parse nice int from char problem/(8 kyu) Parse nice int from char problem.cpp
#include <string>

// 1 Simple solution
int get_age(const std::string& she_said) {
    char age_char = she_said[0];
    int age = age_char - '0';

    return age;
}

// 2 Optimized solution
int get_age(const std::string& she_said) {
    return she_said[0] - '0';
}

// 3 Clever solution
using namespace std;

int get_age(const string& she_said) {
    return stoi(she_said);
}

File: 8kyu/(8 kyu) Parse nice int from char problem/(8 kyu) Parse nice int from char problem.cs
using System;

// 1 Simple solution
public class Kata
{
  public static int GetAge(string inputString)
  {
    char ageChar = inputString[0];
    int age = (int)Char.GetNumericValue(ageChar);
    return age;
  }
}

// 2 Optimized solution
public class Kata
{
  public static int GetAge(string inputString)
  {
    return int.Parse(inputString.Substring(0, 1));
  }
}

// 3 Clever solution
public class Kata
{
  public static int GetAge(string s) => s[0] - '0';
}
